storage/src/storage/sqlite/sync.rs

```diff
@@ -1001,4 +1001,30 @@ impl SyncStorage for SqliteStorage {
         self.init()?;
         Ok(())
     }
+
+    fn trim(&mut self) -> Result<()> {
+        // Remove transactions belonging to non-canon blocks.
+        self.conn.execute(
+            r"
+            DELETE FROM transactions
+            WHERE id IN (
+                SELECT transactions.id
+                FROM transactions
+                INNER JOIN transaction_blocks on transaction_blocks.transaction_id = transactions.id
+                INNER JOIN blocks on blocks.id = transaction_blocks.block_id
```

Comment:
we also probably want to delete transactions with no associated blocks (i.e. was trimmed from legacy impl, but not the transaction) -- or if the transaction found its way in there some other way.

can be done by using
```
LEFT JOIN transaction_blocks on transaction_blocks.transaction_id = transactions.id
LEFT JOIN blocks on blocks.id = transaction_blocks.block_id
WHERE blocks IS NULL OR blocks.canon_height IS NULL -- the second check is sufficient, this is just more obvious/readable with no perf cost
```

Reply:
adjusted the query to use `LEFT JOIN` :+1:; the suggested `blocks IS NULL` check was faulty (it caused no items to get trimmed), so instead I added one for `transaction_blocks.transaction_id` in order to account for orphan locations.